packages/main/src/TabContainer.js

```diff
@@ -496,36 +509,311 @@ class TabContainer extends UI5Element {
 		}
 	}
 
-	_onHeaderBackArrowClick() {
-		this._scrollEnablement.move(-SCROLL_STEP, 0).promise()
-			.then(_ => this._updateScrolling());
-	}
+	async _onEndOverflowButtonClick(event) {
+		const button = this.overflowButton[0] || this.getDomRef().querySelector(".ui5-tc__endOverflowButton > [ui5-button]");
 
-	_onHeaderForwardArrowClick() {
-		this._scrollEnablement.move(SCROLL_STEP, 0).promise()
-			.then(_ => this._updateScrolling());
+		if (event.target !== button) {
+			return;
+		}
+
+		this.items.forEach(item => {
+			item.isInEndOverflow = true;
+		});
+
+		this.responsivePopover = await this._respPopover();
+		if (this.responsivePopover.opened) {
+			this.responsivePopover.close();
+		} else {
+			this.responsivePopover.showAt(button);
+		}
```

Comment:
This code is the same as in the above method. Can you refactor it?

Reply:
Done